sqlmesh/core/context_diff.py

```diff
@@ -185,6 +185,11 @@ def has_changes(self) -> bool:
     def has_snapshot_changes(self) -> bool:
         return bool(self.added or self.removed or self.modified_snapshots)
 
+    @property
+    def added_internal_models(self) -> t.Set[str]:
+        """Returns the set of added internal models."""
+        return {name for name in self.added if not self.snapshots[name].model_kind_name.is_external}
```

Comment:
maybe you should do is_symbolic, or even is_materialized?

Reply:
Why? I thought we only needed to treat external models differently.

Reply:
isn’t it more flexible? we have more options with forward only

Reply:
Yeah it seems like the debate is if embedded or views should be treated like external models in terms of being allowed to be added in forward-only plans.

Reply:
https://github.com/TobikoData/sqlmesh/pull/1130/commits/1e8e9505bef66caef0d870612578ea848a844646